packages/dds/shared-object-base/src/sharedObjectKernel.ts

```diff
@@ -133,6 +133,19 @@ class SharedObjectFromKernel<
 	) {
 		super(id, runtime, attributes, telemetryContextPrefix);
 
+		// This cast is needed since IFluidDataStoreRuntime does not have all the members that its implementation does.
+		// It's a follow-up to a pattern where the property (`minVersionForCollab`) existed on
+		// IFluidDataStoreRuntimeInternalConfig as optional, which allows us to avoid breaking changes to
+		// IFluidDataStoreRuntime by hiding internal members in a separate interface,
+		// but comes at the cost of less compile-time enforcement. Adding an assert now that the period where we needed
+		// to keep `FluidDataStoreRuntime.minVersionForCollab` as optional to maintain back-compat has passed.
```

Comment:
I wasn't 100% convinced about the comment update but can't come up with a better version; feedback here welcome.